public/js/book_library.js

```diff
@@ -0,0 +1,53 @@
+const numberCopies = document.getElementById('numberCopies');
+const numberCopiesError = document.getElementById('numberCopiesError');
+const bookshelf = document.getElementById('bookshelf');
+const bookshelfError = document.getElementById('bookshelfError');
+const section = document.getElementById('section');
+const sectionError = document.getElementById('sectionError');
+const addLibraryBook = document.getElementById('addLibraryBook');
+
+numberCopies.addEventListener('focusout', (e) => {
+  check(numberCopies, numberCopiesError, 'هذا الحقل مطلوب');
+});
+
+bookshelf.addEventListener('focusout', (e) => {
+  check(bookshelf, bookshelfError, 'هذا الحقل مطلوب');
+});
+
+section.addEventListener('focusout', (e) => {
+  check(section, sectionError, 'هذا الحقل مطلوب');
+});
+
+
+addLibraryBook.addEventListener('click', (e) => {
+  const checkNumberCopies = check(numberCopies, numberCopiesError, 'هذا الحقل مطلوب');
+  const checkBookshelf = check(bookshelf, bookshelfError, 'هذا الحقل مطلوب');
+  const checkSection = check(section, sectionError, 'هذا الحقل مطلوب');
+  if (checkNumberCopies && checkBookshelf && checkSection) {
```

Comment:
I think you should send the number of copies to backend and then you can make this process in the backend